packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/EditForm/editForm.js

```diff
@@ -0,0 +1,133 @@
+
+import { mergeClasses } from '../../../../classify';
+import { isRequired } from '../../../../util/formValidators';
+import Button from '../../../Button';
+import Country from '../../../Country';
+import Field, { Message } from '../../../Field';
+import Region from '../../../Region';
+import TextInput from '../../../TextInput';
+import defaultClasses from './editForm.css';
+import EditFormOperations from './editForm.gql';
+
+const EditForm = props => {
+    const { afterSubmit, classes: propClasses, onCancel, shippingData } = props;
+
+    const talonProps = useEditForm({
+        afterSubmit,
+        ...EditFormOperations,
+        shippingData
+    });
+    const { handleSubmit, initialValues, isSaving, isUpdate } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, propClasses);
+
+    const messageComponent = !isUpdate ? (
+        <Message>
+            Set a password at the end of guest checkout to create an account in
+            one easy step.
+        </Message>
+    ) : null;
+
+    const cancelButton = isUpdate ? (
+        <Button
+            classes={{
+                root_normalPriority: classes.submit
+            }}
+            disabled={isSaving}
+            onClick={onCancel}
+            priority="normal"
+        >
+            Cancel
+        </Button>
+    ) : null;
+
+    return (
+        <Form
+            className={classes.root}
+            initialValues={initialValues}
+            onSubmit={handleSubmit}
+        >
+            <Field classes={{ root: classes.field }} id="email" label="Email">
+                <TextInput field="email" validate={isRequired} />
+                {messageComponent}
+            </Field>
+            <Field
+                classes={{ root: classes.firstname }}
+                id="firstname"
+                label="First Name"
+            >
+                <TextInput field="firstname" validate={isRequired} />
+            </Field>
+            <Field
+                classes={{ root: classes.lastname }}
+                id="lastname"
+                label="Last Name"
+            >
+                <TextInput field="lastname" validate={isRequired} />
+            </Field>
+            <Country classes={{ root: classes.field }} validate={isRequired} />
+            <Field
+                classes={{ root: classes.field }}
+                id="street0"
+                label="Street Address"
+            >
+                <TextInput field="street[0]" validate={isRequired} />
+            </Field>
+            <Field
+                classes={{ root: classes.field }}
+                id="street1"
+                label="Street Address 2"
+            >
+                <TextInput field="street[1]" />
+            </Field>
+            <Field classes={{ root: classes.field }} id="city" label="City">
+                <TextInput field="city" validate={isRequired} />
+            </Field>
+            <Region classes={{ root: classes.field }} validate={isRequired} />
```

Comment:
Using `Ukraine` which has no regions and then trying to estimate yields form validation errors. A similar error occurs in the edit form.

[![Image from Gyazo](https://i.gyazo.com/dfaa5690df6fa7c20fa3f64fb6cfa8a7.png)](https://gyazo.com/dfaa5690df6fa7c20fa3f64fb6cfa8a7)

Can we conditionally require region/postcode based on the length of the region response data?

Reply:
This is actually configuration we should pull out of the backend, but we haven't addressed that Story yet about how to properly do that (runtime vs. buildtime etc...). Sometimes region is still required, even though there are no options; sometimes there are options and its not required, each Country can be different. Deferring for now, but create a feature request, this could be a good task for community.

Reply:
https://jira.corp.magento.com/browse/PWA-484